internal/controllers/lvmcluster/resource/lvm_volumegroupnodestatus.go

```diff
@@ -38,67 +42,125 @@ func LVMVGNodeStatus() Manager {
 
 type lvmVGNodeStatus struct{}
 
-// lvmVGNodeStatus unit satisfies resourceManager interface
 var _ Manager = lvmVGNodeStatus{}
 
 func (l lvmVGNodeStatus) GetName() string {
 	return lvmVGNodeStatusName
 }
 
-// EnsureCreated is a noop. This will be created by vg-manager.
 func (l lvmVGNodeStatus) EnsureCreated(r Reconciler, ctx context.Context, cluster *lvmv1alpha1.LVMCluster) error {
+	logger := log.FromContext(ctx).WithValues("resourceManager", l.GetName())
+
+	nodes := v1.NodeList{}
+	if err := r.List(ctx, &nodes); err != nil {
+		return fmt.Errorf("failed to list nodes: %w", err)
+	}
+
+	validNodes, err := selector.ValidNodes(cluster, &nodes)
+	if err != nil {
+		return fmt.Errorf("failed to get valid nodes: %w", err)
+	}
+
+	logger.Info("nodes considered for LVMCluster", "nodes", validNodes, "total", len(nodes.Items))
+
+	for _, node := range validNodes {
+		lvmVGNodeStatus := &lvmv1alpha1.LVMVolumeGroupNodeStatus{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      node.Name,
+				Namespace: r.GetNamespace(),
+			},
+		}
+		result, err := cutil.CreateOrUpdate(ctx, r, lvmVGNodeStatus, func() error {
+			if err := ctrl.SetControllerReference(cluster, lvmVGNodeStatus, r.Scheme()); err != nil {
+				return fmt.Errorf("failed to set controller reference: %w", err)
+			}
+			if !hasDeleteProtectionFinalizer(lvmVGNodeStatus.Finalizers) {
+				lvmVGNodeStatus.SetFinalizers(append(lvmVGNodeStatus.Finalizers, deleteProtectionFinalizer))
+			}
+			return nil
+		})
+
+		if err != nil {
+			return fmt.Errorf("%s failed to reconcile: %w", l.GetName(), err)
+		}
+
+		if result != cutil.OperationResultNone {
+			logger.Info("LVMVolumeGroupNodeStatus applied to cluster", "operation", result, "name", lvmVGNodeStatus.Name)
```

Comment:
Should we add V(2) to the log here to make it consistent with the other resources?

Reply:
Actually I think we have a bug here, I remember writing this some time back but I couldnt get the logs to run with V > 1. With (v1) they still show up in debug but not with V2. Here I forgot the V(2) back in but we probably should go through the code base and change every occurance to V(1) instead and verify the debug logs.

Reply:
Changed to V1. need to open a follow up PR for the rest. Now the logs are visible with `--zap-log-level=debug` in the operator pod